script.js

```diff
@@ -0,0 +1,121 @@
+    let [a, b, c] = [parameters[0], parameters[1], parameters[2]]; // не хотел открывать в эту область видимости(для треугольника — 3 стороны, для квадрата — 1), но кучу строк введения переменных не оно, 'оптимизация' роли не сыграет.
+    let area;
+    switch (figure) {
+        case 'triangle':
+        let hpr = (a + b + c) / 2;
+        area = Math.sqrt(hpr * (hpr-a) * (hpr-b) * (hpr-c));
+        return { area: area, figure: figure, input: { sideA: a, sideB: b, sideC: c }};
+        case 'square':
+        area = a ** 2;
+        return { area: area, figure: figure, input: {sideA: a}};
+        case 'rectangle':
+        area = a * b;
+        return { area: area, figure: figure, input: {sideA: a, sideB: b}};
+        default:
+        throw new Error("Wrong figure");
+    }
+};
+class Human {
+    constructor(name, age, dateOfBirth) {
+        this.name = name;
+        this.age = age;
+        this.dateOfBirth = dateOfBirth;
+    }
+    displayInfo() {
+        return `Name: ${this.name}, Age: ${this.age}, Date of birth: ${this.dateOfBirth}`;
+    }
+}
+class Employee extends Human {
+    constructor(name, age, dateOfBirth, salary, department) {
+        super(name, age, dateOfBirth);
+        this.salary = salary;
+        this.department = department;
+    }
+    displayInfo() {
+        return `${super.displayInfo()}, Salary: ${this.salary}, Department: ${this.department}`;
+    }
+}
+class Manager extends Employee {
+    constructor(name, age, dateOfBirth, salary, department, allTheDevs) {
+        super(name, age, dateOfBirth, salary, department);
+        if (!Array.isArray(allTheDevs) || allTheDevs == undefined) throw new Error("allTheDevs argument must present and must be an array");
+        this.allTheDevs = allTheDevs;
+    }
+    getDevs() {
+        let string = "";
+        this.allTheDevs.forEach((arg, i) => {
+            string = string + `Employee ${i+1} is ${arg}. \n`;
+        });
+        return string;
+    }
+    addDev(dev) {
+        this.allTheDevs.push(dev);
+    }
+    removeDev(index) {
+        this.allTheDevs.splice(index, 1);
+    }
+}
+class Developer extends Employee {
+    constructor(name, age, dateOfBirth, salary, department, manager) {
+        super(name, age, dateOfBirth, salary, department);
+        this.isManagedBy = manager;
+    }
+    changeManager(manager) {
+        this.isManagedBy = manager;
+    }
+    isManagedBy() {
+        return this.isManagedBy;
+    }
+}
+function *generator() {
+    yield {name: prompt("Ваше имя?")};
+    yield {lastname: prompt("Ваша фамилия?")};
+    yield {age: prompt("Ваш возраст?")};
+    yield {profession: prompt("Ваша профессия?")};
+}
+let profile = function() {
+    let obj = {};
+    let a = generator();
+    for (v of a) {
+        let key = Object.keys(v);
+        obj[key] = v[key];
+    }
+    console.log(obj);
+}
+profile();
+let createQueries = function() {
+    let promises = [];
+    rawUrl = "https://jsonplaceholder.typicode.com/users/";
```

Comment:
```suggestion
    const rawUrl = "https://jsonplaceholder.typicode.com/users/";
```
в противном случае rawUrl будет создан на глобальном объекте - а это плохая практика, нельзя его засорять:)

Reply:
забыл добавить const :D